### PR TITLE
Making all tests pass with Implement_for functionality. Enables support most gym/gymnasium versions

### PR DESCRIPTION
**Pull request by vikashplus**
FEATURE: Adding support with passing tests for multiple gym/gymnasium versions
- adding a helper util to prioritize gymnasium import over gym
- standardizing gym_registry_specs for robohive env registration 
- moving all self.np_random.randint to self.np_random.choice as its more standard across versions 
- bugfix in range specifications. high-low<0 for a few instances 
- Adding a buffer (*_) in step and forward to accommodate different number of entities returned 
- starting to use env.unwrapped instead of env.env..... 
- robustifying test_logger. It can effectively fish out the log names now

### CODE REVIEW

**Comment by vikashplus** [comment]
@vmoens @Vittorio-Caggiano @0wu  -- Can I get a round of reviews on this features?
Its largely starts from @vmoens ideas and initial commit on how to make the repo compatible with most gym/gymnasium versions, and builds forward from there.
Big Kudos to Vincent for leading and helping me landing this feature.

File: robohive/envs/claws/__init__.py [hunk 1]
```diff
@@ -38,7 +38,7 @@
         'model_path': curr_dir+'/trifinger/trifinger_reorient.xml',
         'object_site_name': "object",
         'target_site_name': "target",
-        'target_xyz_range': {'high':[.05, .05, 0.9], 'low':[-.05, -.05, 0.99]},
+        'target_xyz_range': {'high':[.05, .05, 0.99], 'low':[-.05, -.05, 0.9]},
```
Comment (vmoens) [inline robohive/envs/claws/__init__.py:41]: I guess this is unrelated?

File: robohive/envs/env_base.py [hunk 2]
```diff
@@ -130,7 +133,7 @@ def _setup(self,
         self._setup_rgb_encoders(self.visual_keys, device=None)
 
         # reset to get the env ready
-        observation, _reward, done, _info = self.step(np.zeros(self.sim.model.nu))
+        observation, _reward, done, *_, _info = self.step(np.zeros(self.sim.model.nu))
```
Comment (vmoens) [inline robohive/envs/env_base.py:136]: We could decorate this too but it's also ok if you don't need the truncated oc

File: robohive/envs/env_base.py [hunk 3]
```diff
@@ -702,7 +728,7 @@ def examine_policy(self,
             ep_rwd = 0.0
             while t < horizon and done is False:
                 a = policy.get_action(o)[0] if mode == 'exploration' else policy.get_action(o)[1]['evaluation']
-                next_o, rwd, done, env_info = self.step(a)
+                next_o, rwd, done, *_, env_info = self.step(a)
```
Comment (vmoens) [inline robohive/envs/env_base.py:731]: Ditto

File: robohive/envs/env_base.py [hunk 4]
```diff
@@ -794,7 +820,7 @@ def examine_policy_new(self,
             ep_rwd = 0.0
 
             # Rollout --------------------------------
-            obs, rwd, done, env_info = self.forward(update_exteroception=True) # t=0
```
Comment (vmoens) [inline robohive/envs/env_base.py:797]: Ditto

File: robohive/envs/env_base.py [hunk 5]
```diff
@@ -825,7 +851,7 @@ def examine_policy_new(self,
 
 
                 # step env using actions from t=>t+1 ----------------------
-                obs, rwd, done, env_info = self.step(act, update_exteroception=True)
+                obs, rwd, done, *_, env_info = self.step(act, update_exteroception=True)
```
Comment (vmoens) [inline robohive/envs/env_base.py:854]: Ditto

**Review by vmoens — APPROVED** [review]
LGTM
there seems to be a bit of unrelated changes for which I have little to say :) 

Have you tried if the gym versions match? We may need to test if things work ok across a bunch of them

File: robohive/envs/env_base.py (hunk 2 above)
Comment (vikashplus) [reply robohive/envs/env_base.py:136]: Seems to work for now. Lets revisit in the future incase this isn't sufficient

**Comment by vikashplus** [comment]
@vmoens  -- I tried something like this - https://github.com/vikashplus/robohive/blob/implement_for/robohive/tests/test_versions.sh 
All the tests succeed. 

Any other salient versions we should check for?

**Review by 0wu — APPROVED** [review]
Stamp for commit

**Review by Vittorio-Caggiano — APPROVED** [review]
LGTM!
Nice work!
